sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/student-scholastic-standings.institutions.controller.ts

```diff
@@ -139,4 +161,85 @@ export class ScholasticStandingInstitutionsController extends BaseController {
       userToken.authorizations.getLocationsIds(),
     );
   }
+
+  /**
+   * Process a text file with applications to be withdrawn.
+   * @param file text file content with all information needed to withdraw applications.
+   * @param validationOnly if true, will execute all validations and return the
+   * errors and warnings. These validations are the same executed during the
+   * final creation process. If not present or false, the file will be processed
+   * and the records will be inserted.
+   * @returns when successfully executed, the list of all withdrawn application ids.
+   * When an error happen it will return all the records (with the error) and
+   * also a user friendly description of the errors to be fixed.
+   */
+  @ApiBadRequestResponse({
+    description:
+      "Error while parsing text file or " +
+      "one or more text data fields received are not in the correct format or " +
+      "there are no records to be imported.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Application withdrawal has invalid data or " +
+      "some error happen with one or more application withdrawal and the entire process was aborted.",
+  })
+  @IsInstitutionAdmin()
+  @IsBCPublicInstitution()
+  @Post("application-bulk-withdrawal")
+  @UseInterceptors(
+    FileInterceptor("file", {
+      limits: uploadLimits(
+        MAX_UPLOAD_FILES,
+        APPLICATION_BULK_WITHDRAWAL_MAX_UPLOAD_PARTS,
+        APPLICATION_BULK_WITHDRAWAL_UPLOAD_MAX_FILE_SIZE,
+      ),
+      fileFilter: textFileFilter,
+    }),
+  )
+  async bulkWithdrawal(
+    @UploadedFile() file: Express.Multer.File,
+    @Query("validation-only", new DefaultValuePipe(false), ParseBoolPipe)
+    validationOnly: boolean,
+  ): Promise<PrimaryIdentifierAPIOutDTO[]> {
+    // Read the entire file content.
+    const fileContent = file.buffer.toString();
+    // Convert the file raw content into text models.
+    let textModels: ApplicationWithdrawalTextModel[];
+    try {
+      textModels =
+        this.applicationWithdrawalImportTextService.readText(fileContent);
+    } catch (error: unknown) {
+      let errorMessage = "Error while parsing text file.";
```

Comment:
I believe we do not need the variable error message. Also we must throw badRequest error only when the error is of custom type.

We do like this in other areas of our application.

```ts
if (
        error instanceof CustomNamedError &&
        error.name === APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR
      ) {
        throw new BadRequestException(
          new ApiProcessError(
            error.message,
            APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR,
          ),
        );
      }
      throw error;

```